### ISSUE TITLE
Guide to generate svg from Inkscape

### ISSUE THREAD

**Issue by PrithviRajNarendra** [wntrblm/Gingerbread#5]
First of all, thanks for this great tool! 👏 

Inkscape is a good tool to generate svg art work which is also open source. Adding a guide to generate svgs from Inkscape similar to what's done for Affinity would be helpful.

**Comment by wraybowling** [comment]
Devoted Inkscape user here. I'm actively trying to understand what's different about my file from the examples.

**Comment by wrickert** [comment]
Yeah, seems like inkscape is not useable for this tool. Thats a shame. 
To add on to what wraybrwling has contributed:
Go into the XML editor (Ctrl+Shift+X) and edit the ID to have the required labels (E.G 'Edge.Cuts', 'F.Cu', etc)
This allows things to show up correctly in Gingerbread but Exporting it causes extremely low polygon images to appear in KiCAD. A 50mm image leaves Inkscape and a 0.75mm images appears (kinda) in KiCAD. 
I have to assume it has to do with DPI mismatch between what Inkscape outputs and what Gingerbread expects. 
I cant find any way to change this in inkscape, even exporting as a plain svg with the DPI setting at 2540 makes no change. 

Any advice or tips would be very appreciated. Gingerbread seems very slick and I would dearly love to incorporate it to my workflow.

**Comment by theacodes** [comment]
I'm happy to accommodate Inkscape designs, but I don't have the time, energy, or desire to fight with Inkscape to figure out its quirks. I don't personally use Inkscape and have very limited, frustrating experience with it. If someone else felt like figuring out what's different between what Inkscape exports and what Gingerbread expects, I'd be happy to made changes to Gingerbread to support it.

**Comment by bradanlane** [comment]
Given that the Svg2Shenzhen plugin for InkScape is not officially "discontinued" and it even links to Gingerbread, I guess it's time for the InkScape+Svg2Shenzhen user base to pool our resources.

_I have had the inverse experience as @theacodes in that my use of Infinity Designer exposed too many bugs (with layers and transformations) to be useable for my recent designs._

Time permitting, I will fork Gingerbread and start looking at the issues starting with layer IDs and then SVG resolution settings.

**Comment by cincodenada** [comment]
Hey y'all, so it looks like @coryalder added support for Inkscape recently, and I just added #15 which improves it a bit and adds instructions to the help page for Inkscape. Gingerbread should now properly recognize layer names from Inkscape.

I haven't run into the resolution issues mentioned, so if y'all could give it a go and report back I'd appreciate it!

**Comment by cincodenada** [comment]
Hmm okay I've been poking around at this and do see the resolution issues. So far the best way to make things happy is to multiply the size you want in mm by 100, and set the dimensions to that number in pixels. So for a 24mm x 36mm file, you'd set page dimensions to 2400px by 3600 px.

I don't like this solution, and have been trying to figure how to handle this in the image loading better, but haven't gotten there yet.

**Comment by bradanlane** [comment]
I attempted to do the "multiplier" method early on and it results is a very messy user experience in InkScape.

The related UX degradation comes from trying to use Inkscape in pixel units vs millimeters. Many / most users who are using Inkscape for KiCad use millimeters as the common unit of measure between the two tools.

I am excited with the work to support Inkscape with Gingerbread. If there is testing or use cases which will help, please let us know.
